src/global/common.rs

```diff
@@ -185,6 +187,138 @@ pub(crate) async fn find_package_records(conda_meta: &Path) -> miette::Result<Ve
     Ok(records)
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[must_use]
+pub(crate) enum StateChange {
+    AddedExposed(String, EnvironmentName),
+    RemovedExposed(String, EnvironmentName),
+    UpdatedExposed(String, EnvironmentName),
+    AddedPackage(PackageRecord, EnvironmentName),
+    AddedEnvironment(EnvironmentName),
+    RemovedEnvironment(EnvironmentName),
+}
+
+impl StateChange {
+    fn env(&self) -> &EnvironmentName {
+        match self {
+            StateChange::AddedExposed(_, env)
+            | StateChange::RemovedExposed(_, env)
+            | StateChange::UpdatedExposed(_, env)
+            | StateChange::AddedPackage(_, env)
+            | StateChange::AddedEnvironment(env)
+            | StateChange::RemovedEnvironment(env) => env,
+        }
+    }
+}
+
+impl fmt::Display for StateChange {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
```

Comment:
This makes sure the reporting is unified across commands

Reply:
I think it would be nice if we could make a formatter that expects an `env_name` and then formats the change a bit more informative. e.g. this is the output of installing python:
```
❯ pixi g i python
✔ Added environment 'python'.
✔ Added package 'python' and installed it with version: '3.12.7' in environment: 'python' .
✔ Exposed '2to3' binary from environment 'python'.
✔ Exposed '2to3-3.12' binary from environment 'python'.
✔ Exposed 'idle3' binary from environment 'python'.
✔ Exposed 'idle3.12' binary from environment 'python'.
✔ Exposed 'pydoc' binary from environment 'python'.
✔ Exposed 'pydoc3' binary from environment 'python'.
✔ Exposed 'pydoc3.12' binary from environment 'python'.
✔ Exposed 'python' binary from environment 'python'.
✔ Exposed 'python3' binary from environment 'python'.
✔ Exposed 'python3-config' binary from environment 'python'.
✔ Exposed 'python3.1' binary from environment 'python'.
✔ Exposed 'python3.12' binary from environment 'python'.
✔ Exposed 'python3.12-config' binary from environment 'python'.
```

Reply:
ChatGPT came with this idea: 
```bash
✔ Added environment 'python' with package 'python' (version: 3.12.7).

✔ Exposed binaries from environment 'python':
   - 2to3, 2to3-3.12
   - idle3, idle3.12
   - pydoc, pydoc3, pydoc3.12
   - python, python3, python3.12
   - python3-config, python3.12-config
  ```